### ISSUE TITLE
Questions About ublock-matches.tsv

### ISSUE THREAD

**Issue by TechnikEmpire** [brave/adblock-rust#556]
Hello,

Running through the `check_matching_equivalent` test, I noticed two filtering rules:

```
- False negative rule: [||googlesyndication.com/safeframe/, (https://tpc.googlesyndication.com/safeframe/1-0-32/html/container.html, https://www.bbc.com/news/science-environment-47150008, other)]
- False negative rule: [||imasdk.googleapis.com^, (https://imasdk.googleapis.com/js/core/bridge3.161.4_en.html, https://www.chip.de/downloads/Minecraft_51705378.html, other)]
```
They match URLs like this:

`https://tpc.googlesyndication.com/safeframe/1-0-32/html/container.html`

Yet, by all accounts (extension, `content-type` header in response etc), these are not `other` types. In fact, your tests, these rules don't create false negatives. They only create false negatives in my unique tests because I'm using heuristics to attempt to better classify arbitrary URLs into appropriate adblock content types and since my heuristics engine returns `document`, they don't match for me.

TL;DR - I'm curious how `ublock-matches.tsv` was generated and what the authoritative method is for deciding that these URLs are `other` rather than `document`.

Thank you.

**Comment by antonok-edm** [comment]
@TechnikEmpire `ublock-matches.tsv` and likewise `requests.json` are (quite old!) test files that have only been used for regression testing in `adblock-rust`. You could consider them to be a historical snapshot of the state of filter lists and some network requests at the time of its creation. I wouldn't rely on them as a "source-of-truth" for anything modern since it's many years out of date at this point!

As per their accuracy as a historical snapshot, I also wouldn't be surprised if there are some minor misclassifications in the data. It was never intended to be 100% accurate, just to provide reasonable coverage of what should/shouldn't be blocked. I don't recall _how_ it was generated at the time (I don't think we have code for it anymore), but it was almost certainly done by crawling and exporting network requests from an automated browser session.

**Comment by TechnikEmpire** [comment]
@antonok-edm Thank you very much. I'll close this out.
